test/Common.FunctionalTests/Utilities/Helpers.cs

```diff
@@ -209,5 +210,26 @@ public static void AllowNoLogs(this IISDeploymentResult deploymentResult)
                 Path.Combine(deploymentResult.DeploymentParameters.PublishedApplicationRootPath, "aspnetcore-debug.log"),
                 "Running test allowed log file to be empty." + Environment.NewLine);
         }
+
+        public static string ReadAllTextFromFile(string filename, ILogger logger)
+        {
+            try
+            {
+                return File.ReadAllText(filename);
+            }
+            catch (Exception ex)
+            {
+                // check if there is a dotnet.exe, iisexpress.exe, or w3wp.exe processes still running.
+                var hostingProcesses = Process.GetProcessesByName("dotnet");
```

Comment:
You use this helper in in-process tests, it needs to support w3wp and iisexpress.